### PR TITLE
Use helm charts for ingress-nginx and cert-manager

### LINKED ISSUE #3: Upgrade to nginx-ingress-controller 0.29.x (or current latest)
It looks like there's a new version available that we should probably upgrade to: https://kubernetes.github.io/ingress-nginx/deploy/

### PR DESCRIPTION
**Pull request by copelco**
Use Helm to manage `ingress-nginx` and `cert-manager`. Both controllers are updated to the latest version. Hopefully, this is the first step towards more maintainable installations. [helm upgrade](https://helm.sh/docs/intro/using_helm/#helm-upgrade-and-helm-rollback-upgrading-a-release-and-recovering-on-failure) has not been tested yet, but the hope is that these helm charts will support upgrades using this role in the future.

* Update release notes for v1.0.0 release
* Use Helm to install `ingress-nginx`
  * `ingress-nginx` controller upgraded from `0.26.1` to `0.44.0` (via `3.23.0` helm chart release)
* Use Helm to install `cert-manager`
  * `cert-manager` controller upgraded from `v0.10.1` to `v1.2.0`
  * The accompanying [caktus.django-k8s](https://github.com/caktus/ansible-role-django-k8s/) role must also be updated to >`v0.0.11` to restore app certificate validation.
* Move Papertrail and New Relic to [caktus.k8s-hosting-services](https://github.com/caktus/ansible-role-k8s-hosting-services)

**BACKWARDS INCOMPATIBLE CHANGE:** To upgrade, see the changelog.

Closes #3
Closes #18

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,8 +65,12 @@ k8s_context: <name of context from ~/.kube/config>
 k8s_cluster_name: <display name for your cluster>
 k8s_letsencrypt_email: <email to contact about expiring certs>
 k8s_echotest_hostname: <test hostname assigned to your cluster ip, e.g. echotest.caktus-built.com>
-# aws only:
-k8s_iam_users: [list of IAM usernames who should be allowed to manage the cluster]
+# AWS only:
```
Comment (tobiasmcnulty) [inline README.md:73]:
```suggestion
# Pin ingress-nginx and cert-manager to current versions so future upgrades of this
# role will not upgrade these charts without your intervention:
# https://github.com/kubernetes/ingress-nginx/releases
k8s_ingress_nginx_chart_version: "3.23.0"
# https://github.com/jetstack/cert-manager/releases
k8s_cert_manager_chart_version: "v1.2.0"
# AWS only:
```

File: README.md [hunk 2]
```diff
@@ -36,7 +36,7 @@ Development sponsored by [Caktus Consulting Group, LLC](http://www.caktusgroup.c
 # file: deploy/requirements.yaml
 
 - src: https://github.com/caktus/ansible-role-k8s-web-cluster
-  version: 0.0.7
+  version: 1.0.0
```
Comment (vkurup) [inline README.md:39]: Yay! Are we planning to use [semantic versioning](https://semver.org/)? If so, can we put a note into the README about that so future devs are sure to update the version in that way.

File: CHANGES.md [hunk 3]
```diff
@@ -4,6 +4,30 @@
...
+
+To upgrade, manually delete the `ingress-nginx` and `cert-manager` namespaces
+and re-deploy this role.
+
+**Other Changes:**
+
+* Move Papertrail and New Relic to
+  [caktus.k8s-hosting-services](https://github.com/caktus/ansible-role-k8s-hosting-services).
+  The existing deployments will not be automatically removed, but they are no
+  longer managed from this role.
```
Comment (vkurup) [inline CHANGES.md]: Maybe add "To take advantage of future changes to those deployments, add the k8s-hosting services role to your requirements.yaml file" (If that is a correct statement)

File: defaults/main.yml [hunk 4]
```diff
@@ -3,14 +3,26 @@ k8s_kubeconfig: ~/.kube/config
...
+# AWS load balancer type.
+# <nlb|elb>
+k8s_aws_load_balancer_type: elb
+
 # <name of context from ~/.kube/config>
 # k8s_context: ''
 
-# k8s_cluster_name will be used in places like a hostname for Papertrail and
-# a display name for NewRelic
-k8s_cluster_name: "{{ k8s_context }}"
```
Comment (vkurup) [inline defaults/main.yml:11]: I think this is still referenced in the README

File: defaults/main.yml [hunk 5]
```diff
@@ -3,14 +3,26 @@ k8s_kubeconfig: ~/.kube/config
 # <aws|gcp|azure|digitalocean>
 # k8s_cluster_type: ''
 
+# DO NOT edit k8s_aws_load_balancer_type after creating your Service. Delete the
+# nginx-ingress namespace and re-run this role if you really need to change the
+# AWS load balancer type.
+# <nlb|elb>
+k8s_aws_load_balancer_type: elb
```
Comment (vkurup) [inline defaults/main.yml]: Looking at the rest of the PR, it seems we want to encourage `nlb` use for future projects? If so, should we make this `nlb` here and update the release notes to tell existing users to set theirs to `elb` if they don't want to upgrade?

**Review by vkurup — COMMENTED** [review]
Looks great. I haven't tried this out because I'm not sure of the best way to do so. Any recommendations for that?

**Comment by tobiasmcnulty** [comment]
What you get when you deploy the most recent ingress-nginx helm chart on top of v0.26.1:

```
$ kubectl -n ingress-nginx get all
NAME                                            READY   STATUS    RESTARTS   AGE
pod/ingress-nginx-controller-74fd5565fb-fprtx   1/1     Running   0          4m23s
pod/nginx-ingress-controller-6d8d7c675b-nzj89   1/1     Running   0          3h54m

NAME                                         TYPE           CLUSTER-IP       EXTERNAL-IP       PORT(S)                      AGE
service/ingress-nginx                        LoadBalancer   10.245.189.95    <snip>    80:30521/TCP,443:30452/TCP   3h53m
service/ingress-nginx-controller             LoadBalancer   10.245.114.46    <snip>   80:31748/TCP,443:32418/TCP   4m23s
service/ingress-nginx-controller-admission   ClusterIP      10.245.113.168   <none>            443/TCP                      4m23s

NAME                                       READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/ingress-nginx-controller   1/1     1            1           4m23s
deployment.apps/nginx-ingress-controller   1/1     1            1           3h54m

NAME                                                  DESIRED   CURRENT   READY   AGE
replicaset.apps/ingress-nginx-controller-74fd5565fb   1         1         1       4m23s
replicaset.apps/nginx-ingress-controller-6d8d7c675b   1         1         1       3h54m
```

**Comment by tobiasmcnulty** [comment]
How to delete cert-manager without breaking the cluster: https://github.com/jetstack/cert-manager/issues/1514#issuecomment-479430080

**Comment by tobiasmcnulty** [comment]
> I haven't tried this out because I'm not sure of the best way to do so. Any recommendations for that?

I added some upgrade instructions which I've tested on Digital Ocean.

Another test might be to deploy just the new ingress to a `ingress-nginx-temp` namespace on Saguaro, and temporarily switch the DNS for a single project to use the new NLB that it creates. Then, change the DNS back and `helm -n ingress-nginx-temp uninstall ingress-nginx && kubectl delete ns ingress-nginx-temp` afterwards to revert the changes (or, continue with the upgrade if everything looks good).

According to my tests on Digital Ocean, it _should_ be possible to do this without downtime...

**Pull request merged**